core/model/src/main/java/org/eclipse/sw360/antenna/model/license/License.java

```diff
@@ -23,83 +34,167 @@
+     * @param properties A map with properties of the license, must not be null.
+     */
     public License(String id, String commonName, String text, Map<String, String> properties) {
         this.id = requireNonNull(id);
         this.commonName = requireNonNull(commonName);
         this.text = requireNonNull(text);
         this.properties = requireNonNull(properties);
     }
 
+    /**
+     * @return The id of the license, normally a SPDX identifier, never null, but could be empty.
+     */
     public String getId() {
         return id;
     }
 
+    /**
+     * @param id The id of the license, normally a SPDX identifier, must not be null.
+     */
     public void setId(String id) {
         this.id = requireNonNull(id);
     }
 
+    /**
+     * @return The name of the license, a more verbal form of the id, never null, but could be empty.
+     */
     public String getCommonName() {
         return commonName;
     }
 
+    /**
+     * @param commonName The name of the license in a more verbal form, must not be null.
+     */
     public void setCommonName(String commonName) {
         this.commonName = requireNonNull(commonName);
     }
 
+    /**
+     * @return The license text of the license as a preformatted text string, never null, but could be empty.
+     */
     public String getText() {
         return text;
     }
 
+    /**
+     * @param text The license text of the license as a formatted text string, must not be null.
+     */
     public void setText(String text) {
         this.text = requireNonNull(text);
     }
 
+    /**
+     * @return The unmodifiable map of properties of the license, never null, but could be empty.
+     */
     Map<String, String> getProperties() {
         return Collections.unmodifiableMap(properties);
     }
 
+    /**
+     * @param properties Replace the current properties by the new property map, if given map is null, an empty map is
+     *                   created.
+     */
     void setProperties(Map<String, String> properties) {
         this.properties = new HashMap<>();
         if(properties != null) {
             this.properties.putAll(properties);
         }
     }
 
+    /**
+     * Set a new property or replace an existing one.
+     *
+     * @param key The key of the property to set, must not be null.
+     * @param property The new value of the property, must not be null.
+     */
     public void setProperty(String key, String property) {
         properties.put(requireNonNull(key), requireNonNull(property));
     }
 
+    /**
+     * Return the value of a property or an empty Optional.
+     *
+     * @param key The key of the queried property, must not be null.
+     * @return The property value stored or an empty Optional.
+     */
     public Optional<String> getProperty(String key) {
-        return Optional.ofNullable(properties.get(key));
+        return Optional.ofNullable(properties.get(requireNonNull(key)))
```

Comment:
This is more than just adding comments.

Reply:
I do not get your point, the commit message also says, fix code smells, the fixes below are doing that, fixing the code smells that were reported by Sonarqube.